test/evp_test.c

```diff
@@ -865,18 +865,20 @@ static int cipher_test_enc(EVP_TEST *t, int enc, size_t out_misalign,
 
     /* Test that the cipher dup functions correctly if it is supported */
     ERR_set_mark();
-    if (EVP_CIPHER_CTX_copy(ctx, ctx_base)) {
-        EVP_CIPHER_CTX_free(ctx_base);
-        ctx_base = NULL;
-    } else {
-        EVP_CIPHER_CTX_free(ctx);
-        ctx = ctx_base;
+    if (!EVP_CIPHER_CTX_copy(ctx, ctx_base)) {
+        TEST_info("Doing a copy of Cipher %s Fails!\n", EVP_CIPHER_get0_name(expected->cipher));
+        ERR_print_errors_fp(stderr);
+        goto err;
```

Comment:
There is an issue with blindly making the test fail if the copy context fails - in our CI we need to check that old FIPS provider still works by running the existing tests. However this will obviously fail because old FIPS providers won't have the dupctx implemented for these ciphers. So the copy context failure needs to be treated as a failure only when run against the current version of FIPS provider, not against older versions.

See the fips_provider_version_match() fips_provider_version_gt(), etc. calls in evp_test.c for other similar cases.

Reply:
copy that, will do.

Reply:
added fips check for failed copy/dup tests